src/components/ims/ims-license.factory.js

```diff
@@ -0,0 +1,187 @@
+/*
+ * CODENVY CONFIDENTIAL
+ * __________________
+ *
+ *  [2015] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+'use strict';
+
+/**
+ * This class is handling the license API retrieval
+ * @author Oleksii Orel
+ */
+export class ImsLicenseApi {
+
+  /**
+   * Default constructor that is using resource
+   * @ngInject for Dependency injection
+   */
+  constructor($resource, $q, $document, $compile) {
+    // keep resource
+    this.$resource = $resource;
+    this.$q = $q;
+    this.$document = $document;
+    this.$compile = $compile;
+
+    // remote call
+    this.remoteLicenseAPI = this.$resource('/im/license', {}, {
+      getProperties: {method: 'GET', url: '/im/license/properties'}
+    });
+
+    // current license
+    this.currentLicense = {
+      key: null,
+      properties: null
+    };
+
+    // default number of allowed users
+    this.defaultNumberOfUsers = 10;
+  }
+
+  /**
+   * Create a nag message element
+   */
+  createNagMessageElement() {
+    if (this.nagMessageElement) {
+      return;
+    }
+    // create nag message element
+    let item = angular.element('<cdvy-nag-message  id="codenvy-nag-message"></cdvy-nag-message>');
```

Comment:
Creation and compilation happens only in case of  ims.isAvailable.
https://github.com/codenvy/dashboard/pull/6/files#diff-e344fe62242ed0193eac3bd13de1c259R75